graph/src/data/query/error.rs

```diff
@@ -182,6 +184,10 @@ impl fmt::Display for QueryExecutionError {
             ScalarCoercionError(_, field, value, scalar_type) => {
                 write!(f, "Failed to coerce value `{}` of field `{}` to scalar type `{}`", value, field, scalar_type)
             }
+            TooComplex(complexity, max_complexity) => {
+                write!(f, "Query has complexity `{}` which is over the maximum `{}`", complexity, max_complexity)
```

Comment:
Users will no doubt see this and "complexity" won't mean much to them. It would improve the user experience if we described its meaning. GitHub returns the following error message:

> "By the time this query traverses to the organizations connection, it is requesting up to 1,000,000 possible nodes which exceeds the maximum limit of 500,000." 

I don't think the message is great, specifically the `organizations connection` part is confusing because there could be many of them.

How about this?

```
Query potentially returns `{}` entities or more and thereby exceeds
the limit of `{}` entities. Possible solutions are reducing the depth
of the query, querying fewer relationships or using `first` to
return smaller collections.
```

Reply:
That's much better!